src/main/java/org/folio/inventory/dataimport/handlers/actions/InstanceUpdateDelegate.java

```diff
@@ -93,13 +99,43 @@ private JsonObject retrieveParsedContent(ParsedRecord parsedRecord) {
     }
   }
 
-  private Future<Instance> updateInstanceInStorage(Instance instance, InstanceCollection instanceCollection) {
+
+  public Future<Instance> updateInstanceInStorageAndRetryIfOLConflictExists(Instance instance, InstanceCollection instanceCollection, Map<String, String> eventPayload, Record marcRecord, Context context) {
     Promise<Instance> promise = Promise.promise();
-    instanceCollection.update(instance, success -> promise.complete(instance),
+
+    instanceCollection.update(instance, success -> {
+        eventPayload.remove(CURRENT_RETRY_NUMBER);
+        promise.complete(instance);
+      },
       failure -> {
-        LOGGER.error(format("Error updating Instance - %s, status code %s", failure.getReason(), failure.getStatusCode()));
-        promise.fail(failure.getReason());
+        if (failure.getStatusCode() == HttpStatus.SC_CONFLICT) {
+          processIfOLErrorExists(eventPayload, marcRecord, context, promise, failure);
+        } else {
+          eventPayload.remove(CURRENT_RETRY_NUMBER);
+          LOGGER.error(format("Error updating Instance - %s, status code %s", failure.getReason(), failure.getStatusCode()));
+          promise.fail(failure.getReason());
+        }
       });
     return promise.future();
   }
+
+  private void processIfOLErrorExists(Map<String, String> eventPayload, Record marcRecord, Context context, Promise<Instance> promise, Failure failure) {
+    int currentRetryNumber = eventPayload.get(CURRENT_RETRY_NUMBER) == null ? 0 : Integer.parseInt(eventPayload.get(CURRENT_RETRY_NUMBER));
+    if (currentRetryNumber < MAX_RETRIES_COUNT) {
+      eventPayload.put(CURRENT_RETRY_NUMBER, String.valueOf(currentRetryNumber + 1));
+      LOGGER.warn("Error updating Instance - {}, status code {}. Retry InstanceUpdateDelegate handler...", failure.getReason(), failure.getStatusCode());
+      handle(eventPayload, marcRecord, context).onComplete(res -> {
+        if (res.succeeded()) {
+          promise.complete();
```

Comment:
please complete the promise with instance to avoid npe in UpdateInstanceQuickMarcEventHandler
https://github.com/folio-org/mod-inventory/blob/e2618e84efc1b20c9508b04dc6aac5890725bdbc/src/main/java/org/folio/inventory/dataimport/handlers/quickmarc/UpdateInstanceQuickMarcEventHandler.java#L32